jobs/etl_jobs/external/metabase/metabase_api.py

```diff
@@ -0,0 +1,158 @@
+import requests
+import json
+from tqdm import tqdm
+from time import sleep
+from google.auth.transport.requests import Request
+from google.oauth2 import id_token
+
+
+class MetabaseAPI:
+    def get_open_id(self, client_id):
+        return id_token.fetch_id_token(Request(), client_id)
+
+    def __init__(self, username, password, host, client_id):
+        self.host = host
+        self.bearer_token = f"Bearer {self.get_open_id(client_id)}"
+
+        url = f"{host}/api/session"
+        response = requests.post(
+            url,
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": self.bearer_token,
+            },
+            data=json.dumps({"username": username, "password": password}),
+        )
+        response.raise_for_status()  # raises exception when not a 2xx response
+        if response.status_code != 204:
+            token_json = response.json()
+            if "id" not in token_json:
+                raise Exception(f"Error login to {host}, error: {token_json}")
+            self.headers = {
+                "Content-Type": "application/json",
+                "X-Metabase-Session": token_json["id"],
+                "Authorization": self.bearer_token,
+            }
+
+    def get_users(self):
+        response = requests.get(f"{self.host}/api/user/", headers=self.headers)
+        return response.json()["data"]
+
+    def put_card(self, _id, _dict):
+        response = requests.put(
+            f"{self.host}/api/card/{_id}", data=json.dumps(_dict), headers=self.headers
+        )
+        return response.json()
+
+    def update_card_collections(self, card_ids, collection_id):
+        params = {"card_ids": card_ids, "collection_id": collection_id}
+
+        response = requests.post(
+            f"{self.host}/api/card/collections",
+            data=json.dumps(params),
+            headers=self.headers,
+        )
+        return response.json()
+
+    def get_cards(self, _id=None):
+        if _id:
+            response = requests.get(f"{self.host}/api/card/{_id}", headers=self.headers)
+        else:
+            response = requests.get(f"{self.host}/api/card/", headers=self.headers)
+        return response.json()
+
+    def format_cards(self, cards):
+        export_cards = []
+        for i, _c in enumerate(cards):
```

Comment:
idem, peux tu spécifier les noms de tes variables d'itérations, ça rend le code plus facile à lire ;)